decidim-bulletin_board-js/src/voter/voter_wrapper_dummy.js

```diff
@@ -8,11 +13,64 @@ export const WAIT_TIME_MS = 500; // 0.5s
 export class VoterWrapper {
   constructor({ voterId }) {
     this.voterId = voterId;
+    this.jointElectionKey = null;
+    this.contests = {};
+  }
+
+  /**
+   * Process the message and update the wrapper status.
+   *
+   * @returns {Object|undefined}
+   */
+  processMessage(messageId, message) {
+    const messageIdentifier = MessageIdentifier.parse(messageId);
+    switch (messageIdentifier.type) {
+      case CREATE_ELECTION: {
+        this.contests = message.description.contests;
+        break;
+      }
+      case END_KEY_CEREMONY: {
+        const content = JSON.parse(message.content);
+        this.jointElectionKey = content.joint_election_key;
+        break;
+      }
+    }
   }
 
-  encrypt(data) {
-    return new Promise((resolve) =>
-      setTimeout(resolve, WAIT_TIME_MS)
-    ).then(() => JSON.stringify(data));
+  async encrypt(vote) {
+    return new Promise((resolve) => setTimeout(resolve, WAIT_TIME_MS)).then(
+      () => {
+        if (!this.jointElectionKey) {
+          console.warn("Invalid election status.");
+          return;
+        }
+        return JSON.stringify(this.createBallot(vote));
+      }
+    );
+  }
+
+  createBallot(vote) {
+    const ballot = { ballot_style: "ballot-style", contests: [] };
+    for (const contestDescription of this.contests) {
+      const contest = {
+        object_id: contestDescription.object_id,
+        ballot_selections: [],
+      };
+      for (const ballotSelection of contestDescription.ballot_selections) {
+        const voted = vote[contestDescription.object_id].includes(
+          ballotSelection.object_id
+        )
+          ? 1
+          : 0;
+        contest.ballot_selections.push({
+          object_id: ballotSelection.object_id,
+          ciphertext:
+            voted + Math.floor(Math.random() * 500) * this.jointElectionKey,
+        });
+      }
+      ballot.contests.push(contest);
+    }
```

Comment:
I'm not a super fan of using `for` and `push` together. Do you think you could rewrite this using two `map` functions? Something like:

```js
ballot.contests = this.contests.map(contestDescription => {
  return {
    // stuff...
    ballot_selections: contestDespcription.ballot_selections.map(ballotSelection => {
      return {
        // more stuff
      };
    }); 
  }
});
```

I tend to find this functional style a bit more readable. What do you think?